### PR TITLE
A tool for conversion of Kraken output into galaxy taxonomy format

### PR DESCRIPTION
**Pull request by nekrut**

### CODE REVIEW

**Comment by bgruening** [comment]
@nekrut what is your take on the https://github.com/galaxyproject/tools-devteam/blob/master/CONTRIBUTING.md. Do we want to have new tools/packages here? Or do we want to collect them under IUC?

File: tool_collections/taxonomy/kraken2tax/kraken2tax.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<tool id="Kraken2Tax" name="Convert Kraken" version="1.0">
+  <description>data to Galaxy taxonomy representation</description>
+    <requirements>
+        <requirement type="package" version="8d245994d7">gb_taxonomy</requirement>
```
Comment (bgruening) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: @erasche @natefoo is awk a standard on all Galaxy supported system? Also syntax wise? Or do we want to depend on gnu_coreutils?

**Comment by nekrut** [comment]
@bgruening I think this is correct. This is my last contribution of a new tool to devtools simply because it is a sort of an update to taxonomy tools. I'll be putting new tool to iuc from now

File: tool_collections/taxonomy/kraken2tax/kraken2tax.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    </requirements>
+  <command>
+<![CDATA[
+
+  awk '{ print \$${read_name}, \$${tax_id} }' OFS="\t" "${input}" | taxonomy-reader \${GALAXY_DATA_INDEX_DIR}/taxonomy/names.dmp \${GALAXY_DATA_INDEX_DIR}/taxonomy/nodes.dmp 1 > "${out_file}"
+
+  ## To function this tool requires names.dmp and nodes.dmp files dowloaded from ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz
+  ## These two files must be placed into a filesystem location accessible at ${GALAXY_DATA_INDEX_DIR}/taxonomy/
```
Comment (bgruening) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: Not sure I like this. Isn't the proper way to either write a data manager for it or use a `.loc` file (maybe even with different versions)?

File: tool_collections/taxonomy/kraken2tax/kraken2tax.xml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+.. class:: warningmark
+
+**Why do I have these "n" things?** 
+
+Be aware that the NCBI taxonomy (ftp://ftp.ncbi.nih.gov/pub/taxonomy/) this tool relies upon is incomplete.  This means that for many species one or more ranks are absent and represented as "**n**". In the above example *subkingdom*, *superphylum* etc. are missing.
+
+
+</help>
+<citations>
```
Comment (bgruening) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: Indentation...

**Comment by bgruening** [comment]
@nekrut I would love to play with these tools asap. 
I have a rough estimation what you are cooking here together and I like it! Metagenomics ftw!

File: tool_collections/taxonomy/kraken2tax/kraken2tax.xml (hunk 1 above)
Comment (nekrut) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: @natefoo? your take?

Comment (bgruening) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: Ok, not `gnu_coreutils` but we have a package for it :) https://github.com/galaxyproject/tools-iuc/tree/master/packages/package_gnu_awk_4_1_0

File: tool_collections/taxonomy/kraken2tax/kraken2tax.xml (hunk 2 above)
Comment (erasche) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: Yep, needs a data manager.

File: tool_collections/taxonomy/kraken2tax/kraken2tax.xml (hunk 1 above)
Comment (erasche) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: definitely use the package.

File: tool_collections/taxonomy/kraken2tax/kraken2tax.xml (hunk 2 above)
Comment (davebx) [inline tool_collections/taxonomy/kraken2tax/kraken2tax.xml]: @bgruening @erasche I've created a data manager, just need to verify that it works with this tool before I create a pull request.

**Comment by davebx** [comment]
See also nekrut#3 and galaxyproject#208

**Comment by nekrut** [comment]
@bgruening @erasche pls, review and accept if you can

**Comment by nekrut** [comment]
and, of course, @nsoranzo

**Comment by bgruening** [comment]
@nekrut you haven't pushed your changes to our requested changes?

**Comment by nekrut** [comment]
@bgruening pls, take another look

**Comment by nekrut** [comment]
@bgruening @nsoranzo @erasche @jmchilton @martenson pls, accept if you think it's ok

**Comment by martenson** [comment]
:+1:

**Pull request merged**